utils/index.d.ts

```diff
@@ -1509,15 +1509,19 @@ export declare function registerReportIssueCommand(commandId: string): void;
  * Registers a namespace that leverages vscode.workspace.fs API to access the file system
  */
 export declare namespace AzExtFsExtra {
-    export function isDirectory(): Promise<boolean>;
-    export function isFile(): Promise<boolean>;
+    export function isDirectory(resource: Uri | string): Promise<boolean>;
+    export function isFile(resource: Uri | string): Promise<boolean>;
     export function ensureDir(resource: Uri | string): Promise<void>;
     export function ensureFile(resource: Uri | string): Promise<void>;
     export function readFile(resource: Uri | string): Promise<string>;
     export function writeFile(resource: Uri | string, contents: string): Promise<void>;
     export function pathExists(resource: Uri | string): Promise<boolean>;
     export function readJSON<T>(resource: Uri | string): Promise<T>
-    export function writeJSON(resource: Uri | string, contents: string | unknown): Promise<void>
+    export function writeJSON(resource: Uri | string, contents: string | unknown, spaces?: string | number): Promise<void>
```

Comment:
Since defaulting to 2 spaces is not the same behavior as [`JSON.stringify`](https://github.com/microsoft/vscode-azuretools/blob/main/azure/src/createAzureClient.ts#L197), it would be good to add a doc comment that at least makes note of that.